### ISSUE TITLE
how to set kernel-parameters

### ISSUE THREAD

**Issue by Ventusfahrer** [pyavitz/debian-image-builder#47]
I used this projekt to build a debian bullseye image for Home Assistant for an Odroid N2+ board (mmc installation).

Everything works fine except that CGroups V2 is default in Bullseye and HomeAssistant did not support V2 (because of Docker??). Getting the necessary kernel-parameter set in the boot-options (systemd.unified_cgroup_hierarchy=false needs to be set) causes me headaches at the moment (I did not get managed it at all up to now).

It would be nice to have it enabled by an option in your configuration would be very helpfull!

kind regards

Peter

**Comment by 0n3man** [comment]
Did you do the HA install using the menu-config, as it includes ability to do full HA setup on top of the image?

**Comment by pyavitz** [comment]
If for whatever reason you didn't use it, those changes can be made by editing; `sudo nano /boot/extlinux/extlinux.conf`

**Comment by Ventusfahrer** [comment]
> Did you do the HA install using the menu-config, as it includes ability to do full HA setup on top of the image?

yes, I used the menu-config Software/HA-Installation step to install an setup home assistant. 

There was another issue with the operating system and I am currently not shure where to place the issue:
iptables and ip6tables do need an

```
update-alternatives --set iptables /usr/sbin/iptables-legacy
update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
```


in order to work. Without it, docker-ce startup will fail

**Comment by Ventusfahrer** [comment]
> If for whatever reason you didn't use it, those changes can be made by editing; `sudo nano /boot/extlinux/extlinux.conf`

really?

the directory /boot/extlinux does not exist on the file system.

therefore I don't know about the file structure of extlinux.conf

What it found was /boot/boot.cmd. I tried to use it and put systemd.unified_cgroup_hierarchy=false into the line

setenv extra "systemd.unified_cgroup_hierarchy=false"

But just rebooting won't do the job. Googling, I found some hints that you have to run mkimage in order to get an updated /boot/boot.scr. But after running mkimage, boot.scr was updated and nothing changed on the /proc/cmdline. 

The last try was to run flash_uboot.sh. After that the operating system refused to load and I reinstalled the whole stuff.

Last shot was to change /boot/config.ini . This one looks more reasonable for the current kernel parameters shown in /proc/cmdline. But again it is not sufficient to only change the file. I need also to find out the right tool-chain to get the changes to the active boot-command.

Any help ist highly appreciated!

Thanks 

Peter

**Comment by pyavitz** [comment]
It would appear you used the petitboot option during img creation. The HA install script doesn't take that build option under consideration, as its more just a novelty and not recommended. You should be able to adjust your cmdline by adding too: `/boot/config.ini`

Example:
https://github.com/pyavitz/debian-image-builder/blob/feature/files/boot/odroid/odroid-config.ini#L7

You don't actually need to edit the boot.cmd, as the config.ini should get sourced by boot.scr. If for whatever reason you do wanna edit it, you can run `bootcmd` to create a new boot.scr when done. Sorry its not really documented in the builder, everyone tends to just use mainline u-boot.

As for iptables, HA recently made some changes requiring me to change a few things in the HA install script, along with additions to the odroid kernel defconfigs. I believe I have got it sorted out, but I still require feedback to be sure.

**Comment by Ventusfahrer** [comment]
curious - that's exactly what I found out in parallel. Fine, so far.

The system was setup by installing the image following: https://magazine.odroid.com/article/os-installation-using-petitboot-and-usb-otg/ because I forgot to purchase an USB-emmc adapter to directly write to the emmc.

I am using the **Direct Boot** option.

My problem was, that for the first try I was using the manual description of https://github.com/home-assistant/supervised-installer. In this case /boot/config.ini is not created at all.

Running the step:

```
wget https://github.com/home-assistant/supervised-installer/releases/latest/download/homeassistant-supervised.deb
dpkg -i homeassistant-supervised.deb
```
will end up in the log:

```
root@7vhas:~# dpkg -i homeassistant-supervised.deb
(Lese Datenbank ... 56046 Dateien und Verzeichnisse sind derzeit installiert.)
Vorbereitung zum Entpacken von homeassistant-supervised.deb ...
.
.
.
--- checkonline.home-assistant.io ping statistics ---
1 packets transmitted, 1 received, 0% packet loss, time 0ms
rtt min/avg/max/mdev = 19.733/19.733/19.733/0.000 ms
[info] Install supervisor startup scripts
[info] Install AppArmor scripts
[info] Start Home Assistant Supervised
[info] Installing the 'ha' cli
[warn] Could not find /etc/default/grub or /boot/firmware/cmdline.txt failed to switch to cgroup v1
[info] Within a few minutes you will be able to reach Home Assistant at:
[info] http://homeassistant.local:8123 or using the IP address of your
[info] machine: http://192.168.221.201:8123
root@7vhas:~#
```
The warning pointing to the grub configuration points out, that the package is not able to fix cgroup v1.

When I run the HA installation via menu-config cgroup v1 is still not configured, but /boot/config.ini is there. 

Configuration after menu-config installed HA:

peter@7vhas:~$ more /boot/config.ini
```
[user_options]
bootlabel=Amlogic
uuid=PARTUUID=06233f8d-02
rootfstype=rootfstype=ext4
verbose=1
usbquirks=usb-storage.quirks=0x1f75:0x0621:u
extra="no_console_suspend clk_ignore_unused"
console="console=tty1 console=ttyAML0,115200n8 console=both"

fk_kvers=Image
initrd=initrd.gz
kernel_addr_r=0x10800000
ramdisk_addr_r=0x30800000
fdtdir=amlogic
fdtfile=meson64_odroidn2_plus.dtb
fdt_addr_r=0x20000000

; Device Tree Overlay
overlay_resize=16384
overlay_profile=
; overlays="spi0 i2c0 i2c1 uart0"

```

```
peter@7vhas:~$ stat -fc %T /sys/fs/cgroup/
cgroup2fs

```
rerun HA deb package to fix cgroup v1

partial output: 
```
[info] Start Home Assistant Supervised
[info] Installing the 'ha' cli
[info] Switching to cgroup v1
got it
[info] Within a few minutes you will be able to reach Home Assistant at:
[info] http://homeassistant.local:8123 or using the IP address of your
[info] machine: http://192.168.221.201:8123
[warn] A reboot is required to apply changes to grub.
```
suggests, that something has changed but /boot/config.ini is not changed and cgroups still v2 configured.

I still have manually to add 
systemd.unified_cgroup_hierarchy=false 
to /boot/config.ini
either to 
`extra="no_console_suspend clk_ignore_unused systemd.unified_cgroup_hierarchy=false"`
or 
`console="console=tty1 console=ttyAML0,115200n8 console=both systemd.unified_cgroup_hierarchy=false"`
line.

So homeassistant-supervised.deb should also see an improvement?

**Comment by Ventusfahrer** [comment]
...maybe it is worth to mention details of the step in menu-config - Software- HA Installation which creates /boot/config.ini somewhere in the documentation so one is able to run it manually?

**Comment by 0n3man** [comment]
I just did verification that installing with uboot on emmc, everything works with these two lines in place using the latest LTS kernel
```
update-alternatives --set iptables /usr/sbin/iptables-legacy
update-alternatives --set ip6tables /usr/sbin/ip6tables-legacy
```
Something most of changed in the LTS kernel as these were not previously required.  [pyavitz](https://github.com/pyavitz) indicated he created an updated kernel configuration for LTS.  I'm going to run a couple test to verify these lines are not required.

I need to do another install to fully verify that as an alternative one can use the 6.0.9 kernel.  I ran that test last night, but I had set the first of the two lines recommended above.  

I could look at modifying the HA install script to update the /boot/config.ini file in cases where petitboot has been selected as the boot loader.   In the event these lines are required with the LTS kernel using the new config I could also add them.

**Comment by 0n3man** [comment]
With the updated defconfig file for odroid-n2 the update-alternatives are lines are no longer required when using the resulting LTS kernel.

**Comment by pyavitz** [comment]
The HA install script now checks for petitboot and presents a notice that the following kernel params should be added to the config.ini.

```sh
systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false usbcore.autosuspend=-1 usbcore.autosuspend=-1 clk_ignore_unused cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory apparmor=1 security=apparmor
```

If it becomes a more popular option "petitboot" in the future, I will automate the process.

**Comment by Ventusfahrer** [comment]
> The HA install script now checks for petitboot and presents a notice that the following kernel params should be added to the config.ini.
> 
> ```shell
> systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false usbcore.autosuspend=-1 usbcore.autosuspend=-1 clk_ignore_unused cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory apparmor=1 security=apparmor
> ```
> 
> If it becomes a more popular option "petitboot" in the future, I will automate the process.

I tested your suggested parameter line, but did not succeed:

here is my config.ini:

```
[user_options]
bootlabel=Amlogic
uuid=PARTUUID=06233f8d-02
rootfstype=rootfstype=ext4
verbose=1
usbquirks=usb-storage.quirks=0x1f75:0x0621:u
;extra="no_console_suspend clk_ignore_unused systemd.unified_cgroup_hierarchy=false"
;extra="no_console_suspend clk_ignore_unused systemd.unified_cgroup_hierarchy=false  systemd.legacy_systemd_cgroup_controller=false"
extra="systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false usbcore.autosuspend=-1 clk_ignore_unused cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory apparmor=1 security=apparmor"
console="console=tty1 console=ttyAML0,115200n8 console=both"

fk_kvers=Image
initrd=initrd.gz
kernel_addr_r=0x10800000
ramdisk_addr_r=0x30800000
fdtdir=amlogic
fdtfile=meson64_odroidn2_plus.dtb
fdt_addr_r=0x20000000

; Device Tree Overlay
overlay_resize=16384
overlay_profile=
; overlays="spi0 i2c0 i2c1 uart0"

```
mount -l | grep cgroup 

shows:

`cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime)
`

When 

```
extra="no_console_suspend clk_ignore_unused systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false"
```
is active, 

`mount - l | grep cgroup 
`

shows the right configuration:

````
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,size=4096k,nr_inodes=1024,mode=755)
cgroup2 on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,name=systemd)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/rdma type cgroup (rw,nosuid,nodev,noexec,relatime,rdma)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)


```

**Comment by pyavitz** [comment]
Update the kernel and see if it makes any difference.

```sh
wget https://github.com/pyavitz/binary/releases/download/060420/linux-headers-meson-g12b_5.15.79-1_arm64.deb
wget https://github.com/pyavitz/binary/releases/download/060420/linux-image-meson-g12b_5.15.79-1_arm64.deb
sudo dpkg -i linux-*.deb
```

**Comment by Ventusfahrer** [comment]
Installation  complanes about downgrading the kernel version :-)
But the version number is one I like more thant th 6.0.8, which looks very strange, if you are used to the numbering schema on ArchLinux or Ubuntu...

```
dpkg: warning: downgrading linux-headers-meson-g12b from 6.0.8-1 to 5.15.79-1
(Reading database ... 56106 files and directories currently installed.)
Preparing to unpack linux-headers-meson-g12b_5.15.79-1_arm64.deb ...
Unpacking linux-headers-meson-g12b (5.15.79-1) over (6.0.8-1) ...
dpkg: warning: downgrading linux-image-meson-g12b from 6.0.8-1 to 5.15.79-1
Preparing to unpack linux-image-meson-g12b_5.15.79-1_arm64.deb ...
Unpacking linux-image-meson-g12b (5.15.79-1) over (6.0.8-1) ...
Setting up linux-headers-meson-g12b (5.15.79-1) ...
Compiling headers ...
^Cdpkg: error processing package linux-headers-meson-g12b (--install):
 installed linux-headers-meson-g12b package post-installation script subprocess was interrupted
Setting up linux-image-meson-g12b (5.15.79-1) ...
update-initramfs: Generating /boot/initrd.img-5.15.79
Generating uInitrd for U-Boot:
Image Name:   initramfs-5.15.79
Created:      Mon Nov 21 15:28:18 2022
Image Type:   ARM Linux RAMDisk Image (gzip compressed)
Data Size:    6966581 Bytes = 6803.30 KiB = 6.64 MiB
Load Address: 00000000
Entry Point:  00000000
Creating initrd.gz.
Errors were encountered while processing:
 linux-headers-meson-g12b

```
the second try installing the packages did succeed.

But the behaviour does not change with the provided kernel:

```
peter@7vhas:~$ cat /proc/cmdline 
console=tty1 console=ttyAML0,115200n8 console=both ro root=PARTUUID=06233f8d-02 net.ifnames=0 rootfstype=ext4 fsck.repair=yes loglevel=1 "systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false usbcore.autosuspend=-1 clk_ignore_unused cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory apparmorextra usb-storage.quirks=0x1f75:0x0621:u rootwait
```
```
peter@7vhas:~$ uname -a
Linux 7vhas 5.15.79 #1 SMP PREEMPT Mon Nov 21 06:44:34 EST 2022 aarch64 GNU/Linux

```
```
peter@7vhas:~$ cat /proc/cmdline
console=tty1 console=ttyAML0,115200n8 console=both ro root=PARTUUID=06233f8d-02 net.ifnames=0 rootfstype=ext4 fsck.repair=yes loglevel=1 "systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false usbcore.autosuspend=-1 clk_ignore_unused cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory apparmorextra usb-storage.quirks=0x1f75:0x0621:u rootwait
```

```
peter@7vhas:~$ mount -l | grep cgroup
cgroup2 on /sys/fs/cgroup type cgroup2 (rw,nosuid,nodev,noexec,relatime)

```

**Comment by pyavitz** [comment]
One kernel version is LTS and the other Stable is all that is.

This is odd:
```sh
^Cdpkg: error processing package linux-headers-meson-g12b (--install):
 installed linux-headers-meson-g12b package post-installation script subprocess was interrupted
```
But if it worked for you the second time, I guess nothing to worry about.

As for the cgroup issue you are still having, for me it resolved its self, after I updated the defconfig(s) and installed a new kernel.

cmdline typo:
wrong: `"systemd.unified_cgroup_hierarchy=false`
correct: `systemd.unified_cgroup_hierarchy=false`

**Comment by pyavitz** [comment]
I just did a fresh install with petitboot and I'm not seeing anything stand out as en error.

```sh
patrick@odroidn2plus:~$ mount -l | grep cgroup
tmpfs on /sys/fs/cgroup type tmpfs (ro,nosuid,nodev,noexec,size=4096k,nr_inodes=1024,mode=755)
cgroup2 on /sys/fs/cgroup/unified type cgroup2 (rw,nosuid,nodev,noexec,relatime,nsdelegate)
cgroup on /sys/fs/cgroup/systemd type cgroup (rw,nosuid,nodev,noexec,relatime,xattr,name=systemd)
cgroup on /sys/fs/cgroup/net_cls,net_prio type cgroup (rw,nosuid,nodev,noexec,relatime,net_cls,net_prio)
cgroup on /sys/fs/cgroup/cpuset type cgroup (rw,nosuid,nodev,noexec,relatime,cpuset)
cgroup on /sys/fs/cgroup/cpu,cpuacct type cgroup (rw,nosuid,nodev,noexec,relatime,cpu,cpuacct)
cgroup on /sys/fs/cgroup/blkio type cgroup (rw,nosuid,nodev,noexec,relatime,blkio)
cgroup on /sys/fs/cgroup/freezer type cgroup (rw,nosuid,nodev,noexec,relatime,freezer)
cgroup on /sys/fs/cgroup/devices type cgroup (rw,nosuid,nodev,noexec,relatime,devices)
cgroup on /sys/fs/cgroup/pids type cgroup (rw,nosuid,nodev,noexec,relatime,pids)
cgroup on /sys/fs/cgroup/rdma type cgroup (rw,nosuid,nodev,noexec,relatime,rdma)
cgroup on /sys/fs/cgroup/perf_event type cgroup (rw,nosuid,nodev,noexec,relatime,perf_event)
cgroup on /sys/fs/cgroup/memory type cgroup (rw,nosuid,nodev,noexec,relatime,memory)
```
```sh
patrick@odroidn2plus:~$ systemctl status
● odroidn2plus
    State: running
     Jobs: 0 queued
   Failed: 0 units
    Since: Tue 2022-11-22 18:57:17 EST; 3min 6s ago
   CGroup: /
           ├─docker 
           │ ├─0a3d09329322355c872bd2774bb57749a17dcdbf35f40f88740458283dd5092a 
           │ │ ├─1958 s6-svscan -t0 /var/run/s6/services
           │ │ ├─2044 s6-supervise s6-fdholderd
           │ │ ├─2214 s6-supervise coredns
           │ │ └─2218 coredns -conf /etc/corefile
           │ ├─9733eb3eb0da78a20a78c1275a0a444805a050dcfe2339988437c2e6bd1b220a 
           │ │ ├─4128 /package/admin/s6/command/s6-svscan -d4 -- /run/service
           │ │ ├─4150 s6-supervise s6-linux-init-shutdownd
           │ │ ├─4152 /package/admin/s6-linux-init/command/s6-linux-init-shutdown>
           │ │ ├─4159 s6-supervise s6rc-fdholder
           │ │ ├─4160 s6-supervise s6rc-oneshot-runner
           │ │ ├─4168 /package/admin/s6/command/s6-ipcserverd -1 -- /package/admi>
           │ │ ├─4196 s6-supervise home-assistant
           │ │ └─4198 python3 -m homeassistant --config /config
           │ ├─f9674c7888cbe8c4c88640bb4df746d10215a9a86150f7002c2ad0c8000701a0 
           │ │ ├─1337 /package/admin/s6/command/s6-svscan -d4 -- /run/service
           │ │ ├─1415 s6-supervise s6-linux-init-shutdownd
           │ │ ├─1418 /package/admin/s6-linux-init/command/s6-linux-init-shutdown>
           │ │ ├─1424 s6-supervise s6rc-fdholder
           │ │ ├─1425 s6-supervise s6rc-oneshot-runner
           │ │ ├─1433 /package/admin/s6/command/s6-ipcserverd -1 -- /package/admi>
           │ │ ├─1470 s6-supervise supervisor
           │ │ ├─1472 s6-supervise watchdog
           │ │ ├─1473 python3 -m supervisor
           │ │ ├─1474 bash /usr/bin/bashio ./run
           │ │ └─1496 sleep 300
           │ ├─ea9bb1419ef6dd426bf0c5ce0f81b38de7cd29b2eed2d61c16e22baa039e1093 
           │ │ ├─2887 s6-svscan -t0 /var/run/s6/services
           │ │ ├─2971 s6-supervise s6-fdholderd
           │ │ ├─3113 s6-supervise observer
           │ │ └─3117 observer
           │ ├─01942a7ba7e66832a69f71b43963b5cbf8b0c36e49ccd892466c508a4bc4bcc2 
           │ │ ├─3217 s6-svscan -t0 /var/run/s6/services
           │ │ ├─3258 s6-supervise s6-fdholderd
           │ │ ├─3407 s6-supervise mdns
           │ │ └─3411 mdns-repeater -f hassio
           │ ├─d93855e2de5d33cf96f1f2eb6220a8916737281f99aa68b64b4ab9f8926a1418 
           │ │ ├─2407 /package/admin/s6/command/s6-svscan -d4 -- /run/service
           │ │ ├─2472 s6-supervise s6-linux-init-shutdownd
           │ │ ├─2474 /package/admin/s6-linux-init/command/s6-linux-init-shutdown>
           │ │ ├─2479 s6-supervise s6rc-fdholder
           │ │ ├─2480 s6-supervise s6rc-oneshot-runner
           │ │ ├─2487 /package/admin/s6/command/s6-ipcserverd -1 -- /package/admi>
           │ │ ├─2689 s6-supervise pulseaudio
           │ │ ├─2691 s6-supervise alsa
           │ │ ├─2693 pulseaudio --system --disallow-exit --exit-idle-time=-1 --d>
           │ │ ├─2696 bash /usr/bin/bashio ./run
           │ │ ├─2708 bash /usr/bin/bashio ./run
           │ │ └─2710 udevadm monitor --subsystem-match=sound
           │ └─b1223be44afe35edf64f65d418582d2890eb3b1a2f7e39a677856164ad9e359e 
           │   ├─1800 /package/admin/s6/command/s6-svscan -d4 -- /run/service
           │   ├─1865 s6-supervise s6-linux-init-shutdownd
           │   ├─1867 /package/admin/s6-linux-init/command/s6-linux-init-shutdown>
           │   ├─1874 s6-supervise s6rc-fdholder
           │   ├─1875 s6-supervise s6rc-oneshot-runner
           │   └─1883 /package/admin/s6/command/s6-ipcserverd -1 -- /package/admi>
           ├─user.slice 
           │ └─user-1000.slice 
           │   ├─user@1000.service 
           │   │ └─init.scope 
           │   │   ├─1623 /lib/systemd/systemd --user
           │   │   └─1624 (sd-pam)
           │   └─session-1.scope 
           │     ├─1541 sshd: patrick [priv]
           │     ├─1732 sshd: patrick@pts/0
           │     ├─1733 -bash
           │     ├─4454 systemctl status
           │     └─4455 pager
           ├─init.scope 
           │ └─1 /sbin/init cgr�\x05\x01�\x7f
           └─system.slice 
             ├─irqbalance.service 
             │ └─546 /usr/sbin/irqbalance --foreground
             ├─haveged.service 
             │ └─528 /usr/sbin/haveged --Foreground --verbose=1
             ├─containerd.service 
             │ ├─ 660 /usr/bin/containerd
             │ ├─1317 /usr/bin/containerd-shim-runc-v2 -namespace moby -id f9674c>
             │ ├─1780 /usr/bin/containerd-shim-runc-v2 -namespace moby -id b1223b>
             │ ├─1938 /usr/bin/containerd-shim-runc-v2 -namespace moby -id 0a3d09>
             │ ├─2386 /usr/bin/containerd-shim-runc-v2 -namespace moby -id d93855>
             │ ├─2868 /usr/bin/containerd-shim-runc-v2 -namespace moby -id ea9bb1>
             │ ├─3196 /usr/bin/containerd-shim-runc-v2 -namespace moby -id 01942a>
             │ └─4109 /usr/bin/containerd-shim-runc-v2 -namespace moby -id 9733eb>
             ├─systemd-udevd.service 
             │ └─383 /lib/systemd/systemd-udevd
             ├─cron.service 
             │ └─537 /usr/sbin/cron -f
             ├─system-serial\x2dgetty.slice 
             │ └─serial-getty@ttyAML0.service 
             │   └─940 /sbin/agetty -o -p -- \u --keep-baud 115200,57600,38400,96>
             ├─docker.service 
             │ ├─ 905 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/cont>
             │ ├─2846 /usr/bin/docker-proxy -proto tcp -host-ip 0.0.0.0 -host-por>
             │ └─2853 /usr/bin/docker-proxy -proto tcp -host-ip :: -host-port 435>
             ├─polkit.service 
             │ └─551 /usr/libexec/polkitd --no-debug
             ├─networkd-dispatcher.service 
             │ └─549 /usr/bin/python3 /usr/bin/networkd-dispatcher --run-startup->
             ├─wpa_supplicant.service 
             │ └─559 /sbin/wpa_supplicant -u -s -O /run/wpa_supplicant
             ├─ModemManager.service 
             │ └─646 /usr/sbin/ModemManager
             ├─haos-agent.service 
             │ └─530 /usr/bin/os-agent
             ├─systemd-journald.service 
             │ └─361 /lib/systemd/systemd-journald
             ├─ssh.service 
             │ └─705 sshd: /usr/sbin/sshd -D [listener] 0 of 10-100 startups
             ├─irq-tweaks.service 
             │ ├─ 953 /bin/bash /usr/bin/irq-tweaks
             │ └─4453 sleep 30
             ├─NetworkManager.service 
             │ └─540 /usr/sbin/NetworkManager --no-daemon
             ├─rsyslog.service 
             │ └─553 /usr/sbin/rsyslogd -n -iNONE
             ├─systemd-resolved.service 
             │ └─511 /lib/systemd/systemd-resolved
             ├─hassio-supervisor.service 
             │ └─1225 docker container wait hassio_supervisor
             ├─udisks2.service 
             │ └─558 /usr/libexec/udisks2/udisksd
             ├─ntp.service 
             │ └─684 /usr/sbin/ntpd -p /var/run/ntpd.pid -g -u 103:109
             ├─dbus.service 
             │ └─538 /usr/bin/dbus-daemon --system --address=systemd: --nofork -->
             ├─system-getty.slice 
             │ └─getty@tty1.service 
             │   └─930 /sbin/agetty -o -p -- \u --noclear tty1 linux
             └─systemd-logind.service 
               └─557 /lib/systemd/systemd-logind
lines 79-138/138 (END)
             ├─irqbalance.service 
             │ └─546 /usr/sbin/irqbalance --foreground
             ├─haveged.service 
             │ └─528 /usr/sbin/haveged --Foreground --verbose=1
             ├─containerd.service 
             │ ├─ 660 /usr/bin/containerd
             │ ├─1317 /usr/bin/containerd-shim-runc-v2 -namespace moby -id f9674c7888cbe8c4c88640bb4df746d10215a9a86150f7002c2ad0c8000701a0 -address /run/containerd/containerd.sock
             │ ├─1780 /usr/bin/containerd-shim-runc-v2 -namespace moby -id b1223be44afe35edf64f65d418582d2890eb3b1a2f7e39a677856164ad9e359e -address /run/containerd/containerd.sock
             │ ├─1938 /usr/bin/containerd-shim-runc-v2 -namespace moby -id 0a3d09329322355c872bd2774bb57749a17dcdbf35f40f88740458283dd5092a -address /run/containerd/containerd.sock
             │ ├─2386 /usr/bin/containerd-shim-runc-v2 -namespace moby -id d93855e2de5d33cf96f1f2eb6220a8916737281f99aa68b64b4ab9f8926a1418 -address /run/containerd/containerd.sock
             │ ├─2868 /usr/bin/containerd-shim-runc-v2 -namespace moby -id ea9bb1419ef6dd426bf0c5ce0f81b38de7cd29b2eed2d61c16e22baa039e1093 -address /run/containerd/containerd.sock
             │ ├─3196 /usr/bin/containerd-shim-runc-v2 -namespace moby -id 01942a7ba7e66832a69f71b43963b5cbf8b0c36e49ccd892466c508a4bc4bcc2 -address /run/containerd/containerd.sock
             │ └─4109 /usr/bin/containerd-shim-runc-v2 -namespace moby -id 9733eb3eb0da78a20a78c1275a0a444805a050dcfe2339988437c2e6bd1b220a -address /run/containerd/containerd.sock
             ├─systemd-udevd.service 
             │ └─383 /lib/systemd/systemd-udevd
             ├─cron.service 
             │ └─537 /usr/sbin/cron -f
             ├─system-serial\x2dgetty.slice 
             │ └─serial-getty@ttyAML0.service 
             │   └─940 /sbin/agetty -o -p -- \u --keep-baud 115200,57600,38400,9600 ttyAML0 vt220
             ├─docker.service 
             │ ├─ 905 /usr/bin/dockerd -H fd:// --containerd=/run/containerd/containerd.sock
             │ ├─2846 /usr/bin/docker-proxy -proto tcp -host-ip 0.0.0.0 -host-port 4357 -container-ip 172.30.32.6 -container-port 80
             │ └─2853 /usr/bin/docker-proxy -proto tcp -host-ip :: -host-port 4357 -container-ip 172.30.32.6 -container-port 80
             ├─polkit.service 
             │ └─551 /usr/libexec/polkitd --no-debug
             ├─networkd-dispatcher.service 
             │ └─549 /usr/bin/python3 /usr/bin/networkd-dispatcher --run-startup-triggers
             ├─wpa_supplicant.service 
             │ └─559 /sbin/wpa_supplicant -u -s -O /run/wpa_supplicant
             ├─ModemManager.service 
             │ └─646 /usr/sbin/ModemManager
             ├─haos-agent.service 
             │ └─530 /usr/bin/os-agent
             ├─systemd-journald.service 
             │ └─361 /lib/systemd/systemd-journald
             ├─ssh.service 
             │ └─705 sshd: /usr/sbin/sshd -D [listener] 0 of 10-100 startups
             ├─irq-tweaks.service 
             │ ├─ 953 /bin/bash /usr/bin/irq-tweaks
             │ └─4453 sleep 30
             ├─NetworkManager.service 
             │ └─540 /usr/sbin/NetworkManager --no-daemon
             ├─rsyslog.service 
             │ └─553 /usr/sbin/rsyslogd -n -iNONE
             ├─systemd-resolved.service 
             │ └─511 /lib/systemd/systemd-resolved
             ├─hassio-supervisor.service 
             │ └─1225 docker container wait hassio_supervisor
             ├─udisks2.service 
             │ └─558 /usr/libexec/udisks2/udisksd
             ├─ntp.service 
             │ └─684 /usr/sbin/ntpd -p /var/run/ntpd.pid -g -u 103:109
             ├─dbus.service 
             │ └─538 /usr/bin/dbus-daemon --system --address=systemd: --nofork --nopidfile --systemd-activation --syslog-only
             ├─system-getty.slice 
             │ └─getty@tty1.service 
             │   └─930 /sbin/agetty -o -p -- \u --noclear tty1 linux
             └─systemd-logind.service 
               └─557 /lib/systemd/systemd-logind
```
```sh
patrick@odroidn2plus:~$ systemctl
  UNIT                                                                                                                                                    LOAD   ACTIVE SUB       DESCRIPTION
  proc-sys-fs-binfmt_misc.automount                                                                                                                       loaded active waiting   Arbitrary Executable File Formats File System Automount Poi>
  sys-devices-platform-soc-ff3f0000.ethernet-net-eth0.device                                                                                              loaded active plugged   /sys/devices/platform/soc/ff3f0000.ethernet/net/eth0
  sys-devices-platform-soc-ff800000.bus-ff803000.serial-tty-ttyAML0.device                                                                                loaded active plugged   /sys/devices/platform/soc/ff800000.bus/ff803000.serial/tty/>
  sys-devices-platform-soc-ffd00000.bus-ffd24000.serial-tty-ttyAML1.device                                                                                loaded active plugged   /sys/devices/platform/soc/ffd00000.bus/ffd24000.serial/tty/>
  sys-devices-platform-soc-ffe05000.sd-mmc_host-mmc1-mmc1:aaaa-block-mmcblk1-mmcblk1p1.device                                                             loaded active plugged   /sys/devices/platform/soc/ffe05000.sd/mmc_host/mmc1/mmc1:aa>
  sys-devices-platform-soc-ffe05000.sd-mmc_host-mmc1-mmc1:aaaa-block-mmcblk1-mmcblk1p2.device                                                             loaded active plugged   /sys/devices/platform/soc/ffe05000.sd/mmc_host/mmc1/mmc1:aa>
  sys-devices-platform-soc-ffe05000.sd-mmc_host-mmc1-mmc1:aaaa-block-mmcblk1.device                                                                       loaded active plugged   /sys/devices/platform/soc/ffe05000.sd/mmc_host/mmc1/mmc1:aa>
  sys-devices-platform-soc-ffe07000.mmc-mmc_host-mmc0-mmc0:0001-block-mmcblk0-mmcblk0boot0.device                                                         loaded active plugged   /sys/devices/platform/soc/ffe07000.mmc/mmc_host/mmc0/mmc0:0>
  sys-devices-platform-soc-ffe07000.mmc-mmc_host-mmc0-mmc0:0001-block-mmcblk0-mmcblk0boot1.device                                                         loaded active plugged   /sys/devices/platform/soc/ffe07000.mmc/mmc_host/mmc0/mmc0:0>
  sys-devices-platform-soc-ffe07000.mmc-mmc_host-mmc0-mmc0:0001-block-mmcblk0-mmcblk0p1.device                                                            loaded active plugged   /sys/devices/platform/soc/ffe07000.mmc/mmc_host/mmc0/mmc0:0>
  sys-devices-platform-soc-ffe07000.mmc-mmc_host-mmc0-mmc0:0001-block-mmcblk0-mmcblk0p2.device                                                            loaded active plugged   /sys/devices/platform/soc/ffe07000.mmc/mmc_host/mmc0/mmc0:0>
  sys-devices-platform-soc-ffe07000.mmc-mmc_host-mmc0-mmc0:0001-block-mmcblk0.device                                                                      loaded active plugged   /sys/devices/platform/soc/ffe07000.mmc/mmc_host/mmc0/mmc0:0>
  sys-devices-platform-soc-ffe09000.usb-ff400000.usb-udc-ff400000.usb.device                                                                              loaded active plugged   /sys/devices/platform/soc/ffe09000.usb/ff400000.usb/udc/ff4>
  sys-devices-platform-soc-ffe09000.usb-ff500000.usb-xhci\x2dhcd.2.auto-usb2-2\x2d1-2\x2d1.3-2\x2d1.3:1.0-host0-target0:0:0-0:0:0:0-block-sda-sda1.device loaded active plugged   SABRENT Storage
  sys-devices-platform-soc-ffe09000.usb-ff500000.usb-xhci\x2dhcd.2.auto-usb2-2\x2d1-2\x2d1.3-2\x2d1.3:1.0-host0-target0:0:0-0:0:0:0-block-sda.device      loaded active plugged   SABRENT
  sys-devices-platform-sound-sound-card0-controlC0.device                                                                                                 loaded active plugged   /sys/devices/platform/sound/sound/card0/controlC0
  sys-devices-virtual-block-ram0.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram0
  sys-devices-virtual-block-ram1.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram1
  sys-devices-virtual-block-ram10.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/ram10
  sys-devices-virtual-block-ram11.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/ram11
  sys-devices-virtual-block-ram12.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/ram12
  sys-devices-virtual-block-ram13.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/ram13
  sys-devices-virtual-block-ram14.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/ram14
  sys-devices-virtual-block-ram15.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/ram15
  sys-devices-virtual-block-ram2.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram2
  sys-devices-virtual-block-ram3.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram3
  sys-devices-virtual-block-ram4.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram4
  sys-devices-virtual-block-ram5.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram5
  sys-devices-virtual-block-ram6.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram6
  sys-devices-virtual-block-ram7.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram7
  sys-devices-virtual-block-ram8.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram8
  sys-devices-virtual-block-ram9.device                                                                                                                   loaded active plugged   /sys/devices/virtual/block/ram9
  sys-devices-virtual-block-zram0.device                                                                                                                  loaded active plugged   /sys/devices/virtual/block/zram0
  sys-devices-virtual-misc-rfkill.device                                                                                                                  loaded active plugged   /sys/devices/virtual/misc/rfkill
  sys-devices-virtual-net-docker0.device                                                                                                                  loaded active plugged   /sys/devices/virtual/net/docker0
  sys-devices-virtual-net-hassio.device                                                                                                                   loaded active plugged   /sys/devices/virtual/net/hassio
  sys-devices-virtual-net-veth5c98acd.device                                                                                                              loaded active plugged   /sys/devices/virtual/net/veth5c98acd
  sys-devices-virtual-net-veth6170366.device                                                                                                              loaded active plugged   /sys/devices/virtual/net/veth6170366
  sys-devices-virtual-net-veth8646041.device                                                                                                              loaded active plugged   /sys/devices/virtual/net/veth8646041
  sys-devices-virtual-net-vethbfdf480.device                                                                                                              loaded active plugged   /sys/devices/virtual/net/vethbfdf480
  sys-devices-virtual-net-vethe8a97a5.device                                                                                                              loaded active plugged   /sys/devices/virtual/net/vethe8a97a5
  sys-devices-virtual-net-vethed971ee.device                                                                                                              loaded active plugged   /sys/devices/virtual/net/vethed971ee
  sys-module-configfs.device                                                                                                                              loaded active plugged   /sys/module/configfs
  sys-module-fuse.device                                                                                                                                  loaded active plugged   /sys/module/fuse
  sys-subsystem-net-devices-docker0.device                                                                                                                loaded active plugged   /sys/subsystem/net/devices/docker0
  sys-subsystem-net-devices-eth0.device                                                                                                                   loaded active plugged   /sys/subsystem/net/devices/eth0
  sys-subsystem-net-devices-hassio.device                                                                                                                 loaded active plugged   /sys/subsystem/net/devices/hassio
  sys-subsystem-net-devices-veth5c98acd.device                                                                                                            loaded active plugged   /sys/subsystem/net/devices/veth5c98acd
  sys-subsystem-net-devices-veth6170366.device                                                                                                            loaded active plugged   /sys/subsystem/net/devices/veth6170366
  sys-subsystem-net-devices-veth8646041.device                                                                                                            loaded active plugged   /sys/subsystem/net/devices/veth8646041
  sys-subsystem-net-devices-vethbfdf480.device                                                                                                            loaded active plugged   /sys/subsystem/net/devices/vethbfdf480
  sys-subsystem-net-devices-vethe8a97a5.device                                                                                                            loaded active plugged   /sys/subsystem/net/devices/vethe8a97a5
  sys-subsystem-net-devices-vethed971ee.device                                                                                                            loaded active plugged   /sys/subsystem/net/devices/vethed971ee                     >
  -.mount                                                                                                                                                 loaded active mounted   Root Mount
  boot.mount                                                                                                                                              loaded active mounted   /boot
  dev-mqueue.mount                                                                                                                                        loaded active mounted   POSIX Message Queue File System
  run-docker-netns-2f12803e3ba8.mount                                                                                                                     loaded active mounted   /run/docker/netns/2f12803e3ba8
  run-docker-netns-3e7bb00718e5.mount                                                                                                                     loaded active mounted   /run/docker/netns/3e7bb00718e5
  run-docker-netns-6cb4c566c51f.mount                                                                                                                     loaded active mounted   /run/docker/netns/6cb4c566c51f
  run-docker-netns-afc4dfb40bde.mount                                                                                                                     loaded active mounted   /run/docker/netns/afc4dfb40bde
  run-docker-netns-default.mount                                                                                                                          loaded active mounted   /run/docker/netns/default
  run-docker-netns-ef4ba059bdd1.mount                                                                                                                     loaded active mounted   /run/docker/netns/ef4ba059bdd1
  run-user-1000.mount                                                                                                                                     loaded active mounted   /run/user/1000
  sys-fs-fuse-connections.mount                                                                                                                           loaded active mounted   FUSE Control File System
  sys-kernel-config.mount                                                                                                                                 loaded active mounted   Kernel Configuration File System
  sys-kernel-debug.mount                                                                                                                                  loaded active mounted   Kernel Debug File System
  tmp.mount                                                                                                                                               loaded active mounted   /tmp
  var-lib-docker-overlay2-2b8a4a690e156a8ee39011fa6eeb62b234b6d2def64f104a8d470a939e00c285-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/2b8a4a690e156a8ee39011fa6eeb62b234>
  var-lib-docker-overlay2-2f16a348f52b787de6518864e43ac12cd8321a7bb8af7579fc729a45bacb2b16-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/2f16a348f52b787de6518864e43ac12cd8>
  var-lib-docker-overlay2-52078782da6ffe1908e37a94049384e2a512125f73ec7b62046877d1d1193c8e-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/52078782da6ffe1908e37a94049384e2a5>
  var-lib-docker-overlay2-544cd1535391379a24840811878cbb5bd6c7dda2cdd18122ac24435f073a9186-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/544cd1535391379a24840811878cbb5bd6>
  var-lib-docker-overlay2-674f9e3dc19468eba198c4be6ba4d81319bd5145e777312d588c6ca616f9d645-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/674f9e3dc19468eba198c4be6ba4d81319>
  var-lib-docker-overlay2-9b2ebc7322de2792bfddb2d18827bba036224a8393f476c9f1c5dd874c0f4f63-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/9b2ebc7322de2792bfddb2d18827bba036>
  var-lib-docker-overlay2-b283e42c8c77122dfd2780f51c134e5a06a0f2e2f416368b63cac132b3469305-merged.mount                                                   loaded active mounted   /var/lib/docker/overlay2/b283e42c8c77122dfd2780f51c134e5a06>
  resolvconf-pull-resolved.path                                                                                                                           loaded active waiting   resolvconf-pull-resolved.path
  systemd-ask-password-console.path                                                                                                                       loaded active waiting   Dispatch Password Requests to Console Directory Watch
  systemd-ask-password-wall.path                                                                                                                          loaded active waiting   Forward Password Requests to Wall Directory Watch          >
  init.scope                                                                                                                                              loaded active running   System and Service Manager
  session-1.scope                                                                                                                                         loaded active running   Session 1 of user patrick                                  >
  alsa-restore.service                                                                                                                                    loaded active exited    Save/Restore Sound Card State
  apparmor.service                                                                                                                                        loaded active exited    Load AppArmor profiles
  console-setup.service                                                                                                                                   loaded active exited    Set console font and keymap
  containerd.service                                                                                                                                      loaded active running   containerd container runtime
  cron.service                                                                                                                                            loaded active running   Regular background program processing daemon
  dbus.service                                                                                                                                            loaded active running   D-Bus System Message Bus
  docker.service                                                                                                                                          loaded active running   Docker Application Container Engine
  fake-hwclock.service                                                                                                                                    loaded active exited    Restore / save the current clock
  getty@tty1.service                                                                                                                                      loaded active running   Getty on tty1
  governor.service                                                                                                                                        loaded active exited    Governor
  haos-agent.service                                                                                                                                      loaded active running   Home Assistant OS Agent
  hassio-apparmor.service                                                                                                                                 loaded active exited    Hass.io AppArmor
  hassio-supervisor.service                                                                                                                               loaded active running   Hass.io supervisor
  haveged.service                                                                                                                                         loaded active running   Entropy Daemon based on the HAVEGE algorithm
  ifplugd.service                                                                                                                                         loaded active exited    LSB: Brings up/down network automatically
  ifupdown-pre.service                                                                                                                                    loaded active exited    Helper to synchronize boot up for ifupdown
  irq-tweaks.service                                                                                                                                      loaded active running   ODROID IRQ Tweaks
  irqbalance.service                                                                                                                                      loaded active running   irqbalance daemon
  keyboard-setup.service                                                                                                                                  loaded active exited    Set the console keyboard layout
  kmod-static-nodes.service                                                                                                                               loaded active exited    Create list of static device nodes for the current kernel
  leds.service                                                                                                                                            loaded active exited    LEDs
  ModemManager.service                                                                                                                                    loaded active running   Modem Manager
  networkd-dispatcher.service                                                                                                                             loaded active running   Dispatcher daemon for systemd-networkd
  networking.service                                                                                                                                      loaded active exited    Raise network interfaces
  NetworkManager-wait-online.service                                                                                                                      loaded active exited    Network Manager Wait Online
  NetworkManager.service                                                                                                                                  loaded active running   Network Manager
  ntp.service                                                                                                                                             loaded active running   Network Time Service
  polkit.service                                                                                                                                          loaded active running   Authorization Manager
  rc-local.service                                                                                                                                        loaded active exited    /etc/rc.local
  resolvconf.service                                                                                                                                      loaded active exited    Nameserver information manager
  rsyslog.service                                                                                                                                         loaded active running   System Logging Service
  serial-getty@ttyAML0.service                                                                                                                            loaded active running   Serial Getty on ttyAML0
  ssh.service                                                                                                                                             loaded active running   OpenBSD Secure Shell server
  systemd-fsck@dev-disk-by\x2duuid-d67eea3c\x2dad37\x2d4c4e\x2da928\x2d9dd70d143739.service                                                               loaded active exited    File System Check on /dev/disk/by-uuid/d67eea3c-ad37-4c4e-a>
  systemd-journal-flush.service                                                                                                                           loaded active exited    Flush Journal to Persistent Storage
  systemd-journald.service                                                                                                                                loaded active running   Journal Service
  systemd-logind.service                                                                                                                                  loaded active running   User Login Management
  systemd-modules-load.service                                                                                                                            loaded active exited    Load Kernel Modules
  systemd-random-seed.service                                                                                                                             loaded active exited    Load/Save Random Seed
  systemd-remount-fs.service                                                                                                                              loaded active exited    Remount Root and Kernel File Systems
  systemd-resolved.service                                                                                                                                loaded active running   Network Name Resolution
  systemd-sysctl.service                                                                                                                                  loaded active exited    Apply Kernel Variables
  systemd-sysusers.service                                                                                                                                loaded active exited    Create System Users
  systemd-tmpfiles-setup-dev.service                                                                                                                      loaded active exited    Create Static Device Nodes in /dev
  systemd-tmpfiles-setup.service                                                                                                                          loaded active exited    Create Volatile Files and Directories
  systemd-udev-trigger.service                                                                                                                            loaded active exited    Coldplug All udev Devices
  systemd-udevd.service                                                                                                                                   loaded active running   Rule-based Manager for Device Events and Files
  systemd-update-utmp.service                                                                                                                             loaded active exited    Update UTMP about System Boot/Shutdown
  systemd-user-sessions.service                                                                                                                           loaded active exited    Permit User Sessions
  tweaks.service                                                                                                                                          loaded active exited    ODROID Specific System Tweaks
  udisks2.service                                                                                                                                         loaded active running   Disk Manager
  user-runtime-dir@1000.service                                                                                                                           loaded active exited    User Runtime Directory /run/user/1000
  user@1000.service                                                                                                                                       loaded active running   User Manager for UID 1000
  wpa_supplicant.service                                                                                                                                  loaded active running   WPA supplicant
  zramswap.service                                                                                                                                        loaded active exited    Linux zramswap setup                                       >
  -.slice                                                                                                                                                 loaded active active    Root Slice
  system-getty.slice                                                                                                                                      loaded active active    system-getty.slice
  system-modprobe.slice                                                                                                                                   loaded active active    system-modprobe.slice
  system-serial\x2dgetty.slice                                                                                                                            loaded active active    system-serial\x2dgetty.slice
  system-systemd\x2dfsck.slice                                                                                                                            loaded active active    system-systemd\x2dfsck.slice
  system.slice                                                                                                                                            loaded active active    System Slice
  user-1000.slice                                                                                                                                         loaded active active    User Slice of UID 1000
  user.slice                                                                                                                                              loaded active active    User and Session Slice                                     >
  dbus.socket                                                                                                                                             loaded active running   D-Bus System Message Bus Socket
  docker.socket                                                                                                                                           loaded active running   Docker Socket for the API
  syslog.socket                                                                                                                                           loaded active running   Syslog Socket
  systemd-fsckd.socket                                                                                                                                    loaded active listening fsck to fsckd communication Socket
  systemd-initctl.socket                                                                                                                                  loaded active listening initctl Compatibility Named Pipe
  systemd-journal-gatewayd.socket                                                                                                                         loaded active listening Journal Gateway Service Socket
  systemd-journald-audit.socket                                                                                                                           loaded active running   Journal Audit Socket
  systemd-journald-dev-log.socket                                                                                                                         loaded active running   Journal Socket (/dev/log)
  systemd-journald.socket                                                                                                                                 loaded active running   Journal Socket
  systemd-rfkill.socket                                                                                                                                   loaded active listening Load/Save RF Kill Switch Status /dev/rfkill Watch
  systemd-udevd-control.socket                                                                                                                            loaded active running   udev Control Socket
  systemd-udevd-kernel.socket                                                                                                                             loaded active running   udev Kernel Socket                                         >
  dev-zram0.swap                                                                                                                                          loaded active active    /dev/zram0                                                 >
  basic.target                                                                                                                                            loaded active active    Basic System
  cryptsetup.target                                                                                                                                       loaded active active    Local Encrypted Volumes
  getty.target                                                                                                                                            loaded active active    Login Prompts
  graphical.target                                                                                                                                        loaded active active    Graphical Interface
  local-fs-pre.target                                                                                                                                     loaded active active    Local File Systems (Pre)
  local-fs.target                                                                                                                                         loaded active active    Local File Systems
  multi-user.target                                                                                                                                       loaded active active    Multi-User System
  network-online.target                                                                                                                                   loaded active active    Network is Online
  network-pre.target                                                                                                                                      loaded active active    Network (Pre)
  network.target                                                                                                                                          loaded active active    Network
  nss-lookup.target                                                                                                                                       loaded active active    Host and Network Name Lookups
  paths.target                                                                                                                                            loaded active active    Paths
  remote-fs.target                                                                                                                                        loaded active active    Remote File Systems
  slices.target                                                                                                                                           loaded active active    Slices
  sockets.target                                                                                                                                          loaded active active    Sockets
  sound.target                                                                                                                                            loaded active active    Sound Card
  swap.target                                                                                                                                             loaded active active    Swap
  sysinit.target                                                                                                                                          loaded active active    System Initialization
  time-set.target                                                                                                                                         loaded active active    System Time Set
  time-sync.target                                                                                                                                        loaded active active    System Time Synchronized
  timers.target                                                                                                                                           loaded active active    Timers
  usb-gadget.target                                                                                                                                       loaded active active    Hardware activated USB gadget                              >
  apt-daily-upgrade.timer                                                                                                                                 loaded active waiting   Daily apt upgrade and clean activities
  apt-daily.timer                                                                                                                                         loaded active waiting   Daily apt download activities
  e2scrub_all.timer                                                                                                                                       loaded active waiting   Periodic ext4 Online Metadata Check for All Filesystems
  fstrim.timer                                                                                                                                            loaded active waiting   Discard unused blocks once a week
  logrotate.timer                                                                                                                                         loaded active waiting   Daily rotation of log files
  man-db.timer                                                                                                                                            loaded active waiting   Daily man-db regeneration
  systemd-tmpfiles-clean.timer                                                                                                                            loaded active waiting   Daily Cleanup of Temporary Directories

LOAD   = Reflects whether the unit definition was properly loaded.
ACTIVE = The high-level unit activation state, i.e. generalization of SUB.
SUB    = The low-level unit activation state, values depend on unit type.
184 loaded units listed. Pass --all to see loaded but inactive units, too.
To show all installed unit files use 'systemctl list-unit-files'.
```
```sh
patrick@odroidn2plus:~$ cat /boot/config.ini
[user_options]
bootlabel=Amlogic
uuid=PARTUUID=38320e9d-02
rootfstype=rootfstype=ext4
verbose=1
usbquirks=usb-storage.quirks=0x1f75:0x0621:u
extra="no_console_suspend clk_ignore_unused systemd.unified_cgroup_hierarchy=false systemd.legacy_systemd_cgroup_controller=false usbcore.autosuspend=-1 cgroup_enable=cpuset cgroup_memory=1 cgroup_enable=memory apparmor=1 security=apparmor"
console="console=tty1 console=ttyAML0,115200n8 console=both"

fk_kvers=Image
initrd=initrd.gz
kernel_addr_r=0x10800000
ramdisk_addr_r=0x30800000
fdtdir=amlogic
fdtfile=meson64_odroidn2_plus.dtb
fdt_addr_r=0x20000000

; Device Tree Overlay
overlay_resize=16384
overlay_profile=
; overlays="spi0 i2c0 i2c1 uart0"
```
```sh
patrick@odroidn2plus:~$ dmesg | grep "Linux version"
[    0.000000] Linux version 5.15.79 (marvin@martian) (gcc-11 (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0, GNU ld (GNU Binutils for Ubuntu) 2.38) #1 SMP PREEMPT Mon Nov 21 06:44:34 EST 2022
```

**Comment by pyavitz** [comment]
This issue was opened back in `Nov 16, 2022` and it is now `May 16, 2023`. I last responded on `Nov 22, 2022` and have yet to get a reply.
